guides/micronaut-data-mongodb/src/test/groovy/example/micronaut/StudentCourseCreationSpec.groovy

```diff
@@ -0,0 +1,80 @@
+package example.micronaut
+
+import example.micronaut.domain.Student
+
+class StudentCourseCreationSpec extends BaseMongoDataSpec {
+
+    def "we begin with no students"() {
+        when:
+        def students = studentClient.list()
+
+        then:
+        students.empty
+    }
+
+    def "and no courses"() {
+        when:
+        def courses = courseClient.list()
+
+        then:
+        courses.empty
+    }
+
+    def "courses can be created"() {
+        when:
+        def math = courseClient.create("Math")
+        def physics = courseClient.create("Physics")
+
+        then:
+        math.name == "Math"
+        math.id != null
+
+        and:
+        physics.name == "Physics"
+        physics.id != null
+
+        and:
+        courseClient.list().size() == 2
+    }
+
+    def "students can be created"() {
+        when:
+        def math = courseClient.create("Math")
```

Comment:
use type

Reply:
out of curiosity, why? Even Java has `val` for local variables now...